### PR TITLE
ls --system and --all flags

### PR DESCRIPTION
**Pull request by bheni**

### CODE REVIEW

**Comment by timsehn** [comment]
I would change this to this spec:
```
~/datasets/irs-new>dolt ls --system
System tables:
	dolt_log
	dolt_docs
	dolt_diff_allagi
	dolt_diff_allnoagi
	dolt_diff_column_mappings
	dolt_history_allagi
	dolt_history_allnoagi
	dolt_history_column_mappings
```
and this:
```
~/datasets/irs-new>dolt ls --system -v
System tables:
	dolt_log
	dolt_docs
	dolt_diff_allagi
	dolt_diff_allnoagi
	dolt_diff_column_mappings
	dolt_history_allagi
	dolt_history_allnoagi
	dolt_history_column_mappings
	(table deleted) dolt_history_noagi
	(branch-1)      dolt_history_zipcodeagi
	(branch-1)      dolt_history_zipcodenoagi
	(table deleted) dolt_diff_noagi
	(branch-2)      dolt_diff_zipcodeagi
	(branch-2)      dolt_diff_zipcodenoagi
```

File: go/cmd/dolt/commands/ls.go [hunk 1]
```diff
@@ -16,23 +16,38 @@ package commands
...
+)
+
+const (
+	systemFlag = "system"
 )
 
 var lsShortDesc = "List tables"
-var lsLongDesc = "Lists the tables within a commit.  By default will list the tables in the current working set" +
-	"but if a commit is specified it will list the tables in that commit."
+var lsLongDesc = "By default will list the tables in the current working set but if a commit is specified it will list " +
```
Comment (zachmu) [inline go/cmd/dolt/commands/ls.go]: s/by default/with no arguments/

File: go/cmd/dolt/commands/ls.go [hunk 2]
```diff
@@ -16,23 +16,38 @@ package commands
...
+	"the tables in that commit.  If the --verbose flag is provided a row count and a hash of the table will also be " +
+	"displayed.\n" +
+	"\n" +
+	"If the --system flag is supplied this will show the dolt system tables which are queryable with SQL.  Some system " +
+	"tables can be queried even if they are not in the working set by specifying appropriate parameters in the SQL " +
+	"queries.  To see these tables too you may pass the --verbose flag.\n" +
+	"\n" +
+	"If the --all flag is supplied user and system tables will be printed."
```
Comment (zachmu) [inline go/cmd/dolt/commands/ls.go]: both user and system

File: go/cmd/dolt/commands/ls.go [hunk 3]
```diff
@@ -16,23 +16,38 @@ package commands
...
 var lsSynopsis = []string{
-	"[<commit>]",
+	"[--options][<commit>]",
```
Comment (zachmu) [inline go/cmd/dolt/commands/ls.go]: space between [] blocks?

File: go/cmd/dolt/commands/ls.go [hunk 4]
```diff
@@ -86,29 +103,49 @@ func (cmd LsCmd) Exec(ctx context.Context, commandStr string, args []string, dEn
...
+			verr = printSystemTables(ctx, root, dEnv.DoltDB, apr.Contains(verboseFlag))
+			cli.Println()
+		}
+
 		return 0
 	}
 
 	cli.PrintErrln(verr.Verbose())
 	return 1
 }
```
Comment (zachmu) [inline go/cmd/dolt/commands/ls.go]: return handleVerrAndExitCode?

Comment (zachmu) [reply go/cmd/dolt/commands/ls.go]: And remove above return 0

File: go/cmd/dolt/commands/ls.go [hunk 5]
```diff
@@ -86,29 +103,49 @@ func (cmd LsCmd) Exec(ctx context.Context, commandStr string, args []string, dEn
...
 			tblNames = append(tblNames, name)
 		}
 	}
 
+	sort.Strings(tblNames)
+
+	return tblNames, nil
+}
+
+func printTables(ctx context.Context, root *doltdb.RootValue, label string, verbose bool) errhand.VerboseError {
```
Comment (zachmu) [inline go/cmd/dolt/commands/ls.go]: printUserTables

**Pull request merged**